spec/models/printables/group_spec.rb

```diff
@@ -82,40 +61,175 @@
+                  data: {
+                    type: 'print_jobs',
+                    attributes: {
+                      printer_name: 'printer1',
+                      label_template_id: 1,
+                      labels: {
+                        body: [
+                          { label: { barcode: '2', barcode2d: '2', top_line: '', middle_line: nil, bottom_line: '' } }
+                        ]
+                      }
+                    }
+                  }
+                }.to_json,
+                headers: post_headers
+              )
+              .to_return(status: 200, body: <<~RESPONSE, headers: { 'Content-Type' => 'application/vnd.api+json' })
+                  {"data":{"type":"print_jobs","attributes":{"printer_name":"printer1","label_template_id":1,"labels":{"body":[{"label":{"barcode":"1","barcode2d":"1","top_line":"","middle_line":null,"bottom_line":""}}]}}}}
+                  RESPONSE
+
+          group.print(config)
+          expect(request).to have_been_made
+        end
+      end
 
-      expect(PMB::PrintJob).to receive(:new)
-        .with(printer_name: 'printer1', label_template_id: 1, labels: { body: body_print })
-        .and_return(saveable_mock)
-      expect(saveable_mock).to receive(:save)
-      group.print(config, 'user1')
+      it 'handles failure with a custom exception' do
+        request =
+          stub_request(:post, "#{uri}/print_jobs").to_return(
+            status: 500,
+            body: <<~RESPONSE,
+                {"data":{"type":"print_jobs","attributes":{"printer_name":"printer1","label_template_id":1,"labels":{"body":[{"label":{"barcode":"2","barcode2d":"2","top_line":"","middle_line":null,"bottom_line":""}},{"label":{"barcode":"1","barcode2d":"1","top_line":"","middle_line":null,"bottom_line":""}}]}}}}
+                RESPONSE
+            headers: {
+              'Content-Type' => 'application/json'
+            }
+          )
+
+        expect { group.print(config) }.to raise_error PrintMyBarcodeJob::PrintingError
+        expect(request).to have_been_made
+      end
     end
 
-    context 'when an asset does not have barcode' do
-      let(:asset1) { create(:asset, facts: props1, barcode: nil) }
-      it 'does not print it' do
-        body_print = [asset2.printable_object]
+    context 'when using v2' do
+      let(:uri) { 'http://localhost:10000/v2' }
+      let(:post_headers) { { 'Content-Type' => 'application/json' } }
+
+      it 'sends the right message to PMB' do
+        request =
+          stub_request(:post, "#{uri}/print_jobs")
+            .with(
+              body: {
+                print_job: {
+                  printer_name: 'printer1',
+                  label_template_name: 'se_ean13_96tube',
+                  labels: [
+                    { label: { barcode: '2', barcode2d: '2', top_line: '', middle_line: nil, bottom_line: '' } },
+                    { label: { barcode: '1', barcode2d: '1', top_line: '', middle_line: nil, bottom_line: '' } }
```

Comment:
Did this body work okay when tested?

For PMB v2 I though an attribute of the label object called "label_name" is required, and that the label isn't needed, but worth checking

e.g. `labels: [ {"barcode": "aBarcode", ..., "label_name": "main_label"}]`

[This](https://github.com/sanger/print_my_barcode/blob/442285ecb96aed30b10965beb7bdeaadd76ab339/app/models/print_job_wrapper.rb#L72) method in PMB checks `label_name` matches the name of the label in PMB

Reply:
Agh, good point.

Reply:
It looks like we actually have two label-name's to worry about, label and label_main. However in practice the 'sqsc_96plate_label_template_code39' template isn't used.

I'll simplify things down by removing the additional templates.